main.tf

```diff
@@ -61,23 +61,23 @@ module "emr-cluster" {
   bootstrap_actions              = var.bootstrap_actions
 
   # Cluster instances
-  subnet_id                   = var.subnet_id
-  key_pair_name               = var.key_pair_name
-  master_instance_group_name  = var.master_instance_group_name
-  master_instance_type        = var.master_instance_type
-  master_group_instance_count = var.master_group_instance_count
-  master_ebs_volumes_count    = var.master_ebs_volumes_count
-  master_ebs_type             = var.master_ebs_type
-  master_ebs_size             = var.master_ebs_size
-  core_instance_group_name    = var.core_instance_group_name
-  core_instance_type          = var.core_instance_type
-  core_group_instance_count   = var.core_group_instance_count
-  core_ebs_volumes_count      = var.core_ebs_volumes_count
-  core_ebs_type               = var.core_ebs_type
-  core_ebs_size               = var.core_ebs_size
-  custom_ami_id               = var.custom_ami_id
-  core_bid_price              = var.core_bid_price
-  master_bid_price            = var.master_bid_price
+  subnet_id                       = var.subnet_id
+  key_pair_name                   = var.key_pair_name
+  master_instance_group_name      = var.master_instance_group_name
+  master_instance_type            = var.master_instance_type
+  master_instance_on_demand_count = var.master_instance_on_demand_count
+  master_ebs_volumes_count        = var.master_ebs_volumes_count
+  master_ebs_type                 = var.master_ebs_type
+  master_ebs_size                 = var.master_ebs_size
+  core_instance_group_name        = var.core_instance_group_name
+  core_instance_type              = var.core_instance_type
+  core_instance_on_demand_count   = var.core_instance_on_demand_count
+  core_ebs_volumes_count          = var.core_ebs_volumes_count
+  core_ebs_type                   = var.core_ebs_type
+  core_ebs_size                   = var.core_ebs_size
+  custom_ami_id                   = var.custom_ami_id
+  core_bid_price                  = var.core_bid_price
+  master_bid_price                = var.master_bid_price
 
```

Comment:
Most (if not all) the new configs should probably be here too, otherwise to get access to them you'll have to use the nested module directly.

Maybe there's a case for excluding a few and being more opinionated at this level of the module. Not the current pattern, but I'm open to the discussion.